### PR TITLE
fix: incorrect error handling of core crypto errors

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

All CoreCrypto errors on decryption are re-thrown as `MLSMessageDecryptionError.failedToDecryptMessage`. This has the effect that we are not ignoring that we should ignore and we are not performing any error recovery on `wrongEpoch`.

### Causes

There's an extra layer to CoreCrypto errors since the introduction of E2EI.

### Solutions

Unpack the `CryptoError` before switching on it.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-data-model/Source/MLS/MLSDecryptionService.swift [hunk 1]
```diff
@@ -154,33 +154,39 @@ public final class MLSDecryptionService: MLSDecryptionServiceInterface {
...
             // Received stale proposal, this proposal is targeting a past epoch and we have already consumed it
-            case CryptoError.StaleProposal: return []
+            case .StaleProposal: return []
 
             // Message arrive in an unmerged group, it has been buffered and will be consumed later.
-            case CryptoError.UnmergedPendingGroup: return []
+            case .UnmergedPendingGroup: return []
+
             default:
                 throw MLSMessageDecryptionError.failedToDecryptMessage
```
Comment (caldrian) [inline wire-ios-data-model/Source/MLS/MLSDecryptionService.swift:184]: We should wrap the actual error in `failedToDecryptMessage`.

File: wire-ios-data-model/Source/MLS/MLSDecryptionService.swift [hunk 2]
```diff
@@ -154,33 +154,39 @@ public final class MLSDecryptionService: MLSDecryptionServiceInterface {
...
             }
+        } catch {
+            WireLogger.mls.error("failed to decrypt message for group (\(groupID.safeForLoggingDescription)) and subconversation type (\(String(describing: subconversationType))): \(String(describing: error)) | \(debugInfo)")
+
+            throw MLSMessageDecryptionError.failedToDecryptMessage
```
Comment (caldrian) [inline wire-ios-data-model/Source/MLS/MLSDecryptionService.swift:189]: Also here, the case `failedToDecryptMessage` could wrap the actual error.

**Review by caldrian — APPROVED** [review]

File: wire-ios-data-model/Source/MLS/MLSDecryptionService.swift (hunk 1 above)
Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSDecryptionService.swift:184]: Will do in follow up PR

File: wire-ios-data-model/Source/MLS/MLSDecryptionService.swift (hunk 2 above)
Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSDecryptionService.swift:189]: Same as above

**Pull request merged**